src/backends/libtorch/torch_backend.cc

```diff
@@ -0,0 +1,608 @@
+    const auto& cc_itr = Config().cc_model_filenames().find(cc);
+    cc_model_filename = (cc_itr == Config().cc_model_filenames().end())
+                            ? Config().default_model_filename()
+                            : cc_itr->second;
+  }
+
+  const auto& lp_itr = paths.find(cc_model_filename);
+  if (lp_itr == paths.end()) {
+    return Status(
+        RequestStatusCode::INTERNAL, "unable to find LibTorch model '" +
+                                         cc_model_filename + "' for " + Name());
+  }
+
+  if (gpu_device == Context::NO_GPU_DEVICE) {
+    LOG_INFO << "Creating instance " << instance_name << " on CPU using "
+             << cc_model_filename;
+  } else {
+    LOG_INFO << "Creating instance " << instance_name << " on GPU "
+             << gpu_device << " (" << cc << ") using " << cc_model_filename;
+  }
+
+  // Max batch size. A value of 0 in the config becomes NO_BATCHING.
+  const int mbs = (Config().max_batch_size() <= 0) ? Context::NO_BATCHING
+                                                   : Config().max_batch_size();
+
+  contexts_.emplace_back(instance_name, gpu_device, mbs);
+  Context& context = contexts_.back();
+
+  // If this is a sequence model then add the required inputs...
+  if (Config().has_sequence_batching()) {
+    RETURN_IF_ERROR(ValidateSequenceControl(
+        ModelSequenceBatching::Control::CONTROL_SEQUENCE_START, &input_names));
+    RETURN_IF_ERROR(ValidateSequenceControl(
+        ModelSequenceBatching::Control::CONTROL_SEQUENCE_READY, &input_names));
+  }
+
+  try {
+    LibTorchWorkspace* ltws;
+    LibTorchWorkspace::Error err = LibTorchWorkspaceCreate(
+        &ltws, Config().name(), Config().max_batch_size(), input_names,
+        output_names, gpu_device, lp_itr->second);
+    if (!err.IsOk()) {
+      return Status(RequestStatusCode::INTERNAL, err.Message());
+    }
+
+    context.torch_model_ = ltws.torch_model_;
+  }
+  catch (const std::exception& ex) {
+    return Status(
+        RequestStatusCode::INTERNAL,
+        "load failed for '" + Config().name() + "': " + ex.what());
+  }
+
+  RETURN_IF_ERROR(context.ValidateInputs(Config().input()));
+  RETURN_IF_ERROR(context.ValidateOutputs(Config().output()));
+
+  return Status::Success;
+}
+
+Status
+LibTorchBackend::ValidateSequenceControl(
+    const ModelSequenceBatching::Control::Kind control_kind,
+    std::vector<std::string>* input_names)
+{
+  std::string tensor_name;
+  RETURN_IF_ERROR(GetSequenceControlProperties(
+      Config().sequence_batching(), Name(), control_kind, true /* required */,
+      &tensor_name, nullptr, nullptr, nullptr, nullptr, nullptr));
+  input_names->push_back(tensor_name);
+
+  return Status::Success;
+}
+
+Status
+LibTorchBackend::Context::ValidateInputs(
+    const ::google::protobuf::RepeatedPtrField<ModelInput>& ios)
+{
+  for (const auto& io : ios) {
+    // For now, skipping the check if potential names is empty
```

Comment:
This comment doesn't make sense to me. It seems you are validating that the datatype is supported.

Reply:
Removed